packages/sources/universal-v3/src/endpoint/lambda.ts

```diff
@@ -0,0 +1,203 @@
+
+export const inputParameters: InputParameters = {
+  source: {
+    description: 'JavaScript source code to be executed',
+    type: 'string',
+    required: true,
+  },
+  subscriptionId: {
+    description: 'Subscription Id used by the request',
+    required: true,
+  },
+  requestId: {
+    description: 'Request Id unique to each request',
+    required: true,
+  },
+  maxResponseBytes: {
+    description: 'Maximum size of response data, in bytes',
+  },
+  numAllowedQueries: {
+    description: 'Number of HTTP queries that can be performed in a request',
+  },
+  secrets: {
+    description: 'Base64 bytestring representing an encrypted secrets object in base64 format',
+    dependsOn: ['secretsOwner', 'subscriptionId'],
+  },
+  secretsOwner: {
+    description: 'Owner of the encrypted secrets used for signature verification',
+    dependsOn: ['secrets', 'subscriptionId'],
+  },
+  args: {
+    description: 'Array of on-chain arguments which are passed to the source code',
+    type: 'array',
+  },
+  codeLocation: {
+    description:
+      'Location of user-provided code, encoded as an integer represeting a enum (0 = on-chain)',
+    type: 'number',
+    options: [0],
+  },
+  secretsLocation: {
+    description:
+      'Location of user-provided secrets, encoded as an integer represeting a enum (0 = on-chain)',
+    type: 'number',
+    options: [0],
+  },
+  language: {
+    description:
+      'Language of the user-provided code, encoded as an integer represeting a enum (0 = javascript)',
+    type: 'number',
+    options: [0],
+  },
+}
+
+interface Input {
+  source: string
+  subscriptionId: string
+  requestId: string
+  numAllowedQueries?: number
+  maxResponseBytes?: number
+  args?: string[]
+  secrets?: base64bytes
+  secretsOwner?: string
+  codeLocation?: Location.Onchain
+  secretsLocation?: Location.Onchain
+  language?: Language.JavaScript
+}
+
+type base64bytes = string
+
+enum Location {
+  Onchain = 0,
+}
+
+enum Language {
+  JavaScript = 0,
+}
+
+interface Request extends RequestGenerics {
+  Params: Input
+}
+
+interface AdapterIO extends TransportGenerics {
+  Request: Request
+  Response: Response
+  CustomSettings: typeof customSettings
+}
+
+interface LambdaTransport extends Transport<AdapterIO> {
+  lambdaController: LambdaController
+}
+
+class Lambda implements LambdaTransport {
+  lambdaController!: LambdaController
+
+  initialize = async (
+    _dependencies: TransportDependencies<AdapterIO>,
+    config: AdapterConfig<typeof customSettings>,
+  ): Promise<void> => {
+    this.lambdaController = new LambdaController(config)
+    await this.lambdaController.initalize()
+    return
+  }
+
+  // This function handles each request sent to the EA
+  foregroundExecute = async (
+    req: AdapterRequest<Request>,
+    config: AdapterConfig<typeof customSettings>,
+  ): Promise<AdapterResponse<Response>> => {
+    const requestId = req.requestContext.data.requestId
```

Comment:
More readable to destructure? ¯\_(ツ)_/¯
```suggestion
    const { requestId, source, subscriptionId, secrets, secretsOwner, args, numAllowedQueries, maxResponseBytes} = req.requestContext.data
```